graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -12,10 +12,41 @@
 #include "../index_select.h"
 #include "./common.h"
 #include "./utils.h"
+#include "cub/cub.cuh"
 
 namespace graphbolt {
 namespace ops {
 
+std::pair<torch::Tensor, torch::Tensor> Sort(
+    torch::Tensor input, int num_bits) {
+  int64_t n_items = input.size(0);
+  auto orig_idx = torch::arange(n_items, input.options().dtype(torch::kLong));
+  auto sorted_array = torch::empty_like(input);
+  auto sorted_idx = torch::empty_like(orig_idx);
+  cuda::CUDAWorkspaceAllocator allocator;
+  AT_DISPATCH_INDEX_TYPES(
+      input.scalar_type(), "SortImpl", ([&] {
+        using IdType = index_t;
+        const auto keys_in = input.data_ptr<index_t>();
```

Comment:
just to confirm: input.data_ptr is actual input_keys, and original_idx is actually input value? The name looks weird to me.

Reply:
We are sorting the input indices, but want to keep track of their initial positions. Thus, we initialize orig_idx to torch::arange so that when we sort the arrays together, sorted_idx contains where the output should go in the output.

Reply:
We sort the input along with [0, 1, 2, ....] together. As a result, the sorted_values gives us where the original positions are of the sorted items before they were sorted.